File: app/src/main/java/com/yegor/tesonet/partyapp/ui/list/ServersListAdapter.java
package com.yegor.tesonet.partyapp.ui.list;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.yegor.tesonet.partyapp.R;
import com.yegor.tesonet.partyapp.model.Server;

import java.util.ArrayList;
import java.util.List;

/**
 * Adapter for servers list
 */
class ServersListAdapter extends RecyclerView.Adapter<ServerViewHolder> {

    private Context mContext;
    private List<Server> mServers = new ArrayList<>();

    ServersListAdapter(Context context) {
        mContext = context;
    }

    @Override
    public ServerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.servers_list_item, parent, false);
        return new ServerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ServerViewHolder holder, int position) {
        Server server = mServers.get(position);
        holder.name.setText(server.getName());
        holder.distance.setText(getDistanceRepresentation(server.getDistance()));
    }

    @Override
    public int getItemCount() {
        return mServers.size();
    }

    void setServers(List<Server> servers) {
        mServers = servers;
        notifyDataSetChanged();
    }

    private String getDistanceRepresentation(String distance){
        return distance + " km";
    }
}

File: app/src/main/java/com/tesonet/testio/ui/serverslist/ServersListAdapter.kt
package com.tesonet.testio.ui.serverslist

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.tesonet.testio.R
import com.tesonet.testio.data.local.entity.Server
import kotlinx.android.synthetic.main.item_serverslist.view.*


class ServersListAdapter : RecyclerView.Adapter<ServersListAdapter.ViewHolder>() {

    var servers = emptyList<Server>()

    class ViewHolder(val layout: ViewGroup) : RecyclerView.ViewHolder(layout)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layout = LayoutInflater.from(parent.context).inflate(R.layout.item_serverslist, parent, false) as ViewGroup
        return ViewHolder(layout)
    }

    override fun getItemCount() = servers.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.layout.server.text = servers[position].name
        holder.layout.distance.text = holder.layout.context.getString(R.string.km, servers[position].distance)
    }
}